src/components/Carousel/components/Slide/Slide.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright 2019, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from '@emotion/styled';
+import { css, keyframes } from '@emotion/core';
+
+import { childrenPropType } from '../../../../util/shared-prop-types';
+
+import * as SlideService from './SlideService';
+import { ANIMATION_DURATION, SLIDE_DIRECTIONS } from '../../constants';
+
+const baseStyles = ({ index, stackOrder, width }) => css`
+  label: carousel__slide;
+  width: 100%;
+  flex-grow: 0;
+  flex-shrink: 0;
+  flex-basis: ${width};
+  transform: translate3d(${-index * 100}%, 0, 0);
+  backface-visibility: hidden;
+  position: relative;
+  z-index: ${stackOrder};
+`;
+const Wrapper = styled('div')(baseStyles);
+
+const slideIn = keyframes`
+  from {
+    width: 0%;
+  }
+  to {
+    width: 100%;
+  }
+`;
+const slideOut = keyframes`
+  from {
+    width: 100%;
+  }
+  to {
+    width: 0%;
+  }
+`;
+const animationStyles = ({
+  theme,
+  isAnimating,
+  animationDuration,
+  animationName
+}) => css`
+  box-shadow: 0 0 1px rgba(0, 0, 0, 0.05);
+  overflow: hidden;
+  will-change: width;
+  ${isAnimating &&
+    css`
+      animation-name: ${animationName};
+      animation-duration: ${animationDuration}ms;
+      animation-fill-mode: forwards;
+      animation-timing-function: ${theme.transitions.easeInOutCubic};
+    `};
```

Comment:
```suggestion
      animation-name: ${animationName};
      animation-duration: ${animationDuration}ms;
      animation-fill-mode: forwards;
      animation-timing-function: ${theme.transitions.easeInOutCubic};
      animation-play-state: ${isAnimating ? 'running' : 'paused'}
```

Maybe we can use the play-state here? This way you don't have to conditionally add rules, just one value

Reply:
Due to the way `animation-play-state` prop works, it leads to inconsistent sliding animations -
> Resuming a paused animation will start the animation from where it left off at the time it was paused, rather than starting over from the beginning of the animation sequence.